### PR TITLE
FileSystemLibrary: Init

### LINKED ISSUE #29: Common Utility Library
This is the list of utility methods that can be reused by both FileMapper and FileSystem. @muskangarg98 please add any functions if any on this issue. Even If anticipating any API which is not yet implemented but can be useful for both the modules, then we can add it here. 

Let us tick, once we have add the functionality the library 

- [x] `getFullDrivePath()` (Implemented by @muskangarg98)
- [x] `isValidAbsolutePath()` (Implemented by @satviksr)
- [x] `getFileSystemType()`(Implemented by @satviksr)

> Lets put only methods which require no state here

### PR DESCRIPTION
**Pull request by satviksr**
Fixes #29 
- [X] Local File Path APIs - @satviksr 
- [X] Drive Util API - @muskangarg98 
- [X] Error Types - @muskangarg98 
- [x] Sanitize Path - @satviksr 
- [X] Block Prompt Decorator - @satviksr

### CODE REVIEW

**Comment by satviksr** [comment]
@muskangarg98 Can you add the Error classes. I have added everything from my side.

**Comment by muskangarg98** [comment]
@satviksr Can we name it something more meaningful, something like "Shared Library" ? 
(or anything else you want to suggest)

**Comment by satviksr** [comment]
> @satviksr Can we name it something more meaningful, something like "Shared Library" ?
> (or anything else you want to suggest)

How about FileSystemSharedLibrary?

File: FileSystem/Util.js [hunk 1]
```diff
@@ -210,44 +210,3 @@ function deleteFileIfExists(localPath) {
...
-      }
-      if (err instanceof PromptException) {
-        var timeOutMessage =
-            'Current macro execution has timed out.\nPlease try again.';
-        SpreadsheetApp.getUi().alert('Execution Failed', timeOutMessage);
-      }
-      throw err;
-    }
-  }
-}
```
Comment (satviksr) [inline FileSystem/Util.js]: Moved to Shared Library

**Review by monikadhok — APPROVED** [review]

**Review by paneendra-google — APPROVED** [review]

**Pull request merged**